cloud/errors.go

```diff
@@ -18,12 +18,18 @@ package azure
 
 import (
 	"errors"
-
 	"github.com/Azure/go-autorest/autorest"
 )
 
-// ResourceNotFound parses the error to check if it's a resource not found
+var ErrNotOwned = errors.New("resource is not managed and cannot be deleted")
+
+// ResourceNotFound parses the error to check if it's a resource not found error.
 func ResourceNotFound(err error) bool {
 	derr := autorest.DetailedError{}
 	return errors.As(err, &derr) && derr.StatusCode == 404
 }
+
+// ResourceNotOwned parses the error to check if it's a resource not owned error.
+func ResourceNotOwned(err error) bool {
```

Comment:
Did you specifically choose to write this method rather than using `errors.Is(err, ErrNotOwned)`?

Might be nice to use them uniformly like, `errors.Is(err, ResourceNotFound)` and `errors.Is(err, ResourceNotOwned)`.

Reply:
no, I didn't think of using errors.Is, I guess I just tried to be consistent with what was already there for not found. I can change it, not sure if it will work for NotFound too though (?)

Reply:
I think it would work for not found, but we would need to create an error type in our package which would wrap the inner not found error. Then implement `func (e *CloudError) Is(target error) bool {...}` where cloud error is that wrapping type.

Maybe this should be part of an error refactoring, rather than this PR.

Just wanted to bring it up, but I'm not super passionate about it.